agent_app_sample_code/00_shared_config.py

```diff
@@ -0,0 +1,95 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC ## Shared configuration
+# MAGIC
+# MAGIC This notebook initializes a `AgentCookbookConfig` Pydantic class to define parameters that are shared between the cookbook notebooks:
+# MAGIC - Unity Catalog schema 
+# MAGIC - MLflow Experiment to track versions of the Agent and their associated quality/cost/latency evaluation results
+# MAGIC - Unity Catalog model that stores versions of the Agent's code/config that are deployed
+# MAGIC - Evaluation Set Delta Table
+# MAGIC
+# MAGIC This notebook does the following:
+# MAGIC 1. Creates the UC catalog/schema if they don't exist
+# MAGIC 2. Serializes the configuration to `config/cookbook_config.yaml` so other notebooks can load it
+# MAGIC
+# MAGIC **Important: We suggest starting from a fresh clone of the cookbook if you need to change this configuration.**
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC **Important note:** Throughout this notebook, we indicate which cells you:
+# MAGIC - ✅✏️ *should* customize - these cells contain config settings to change
+# MAGIC - 🚫✏️ *typically will not* customize - these cells contain boilerplate code required to validate / save the configuration
+# MAGIC
+# MAGIC *Cells that don't require customization still need to be run!*
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### 🚫✏️ Install Python libraries
+
+# COMMAND ----------
+
+# MAGIC %pip install -qqqq -U -r requirements.txt
+# MAGIC dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### 🚫✏️ Get user info to set default values
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+
+# Get current user's name & email 
+w = WorkspaceClient()
+user_email = w.current_user.me().user_name
+user_name = user_email.split("@")[0].replace(".", "_")
+
+# Get the workspace default UC catalog
+default_catalog = spark.sql("select current_catalog() as cur_catalog").collect()[0]['cur_catalog']
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### ✅✏️ Configure this instance of the cookbook
+
+# COMMAND ----------
+
+from cookbook_utils.cookbook_config import AgentCookbookConfig
+
+shared_config = AgentCookbookConfig(
+    uc_catalog_name=f"{default_catalog}",
+    uc_schema_name=f"{user_name}_agents", 
+    uc_asset_prefix="agent_app_name", # Prefix to every created UC asset, typically the Agent's name
```

Comment:
Unclear how to scale this to multiple models - IMO schema should be the “agent app name” (where agent apps contain one or more models/data sources)

Reply:
That would simplify the confusing idea of us_asset_prefix...and avoid long names for each asset

Reply:
Is that the common pattern you see @FMurray - one agent == one schema?

Reply:
By agent app - I mean one use case, which might have several agent models, like for example 3 distinct RAG models and an supervisor model. 

I see customers who have less permissive UC setups typically apply grants on the schema (occasionally on catalog), so I wouldn't want to have one schema per model, because that would slow them down.